### PR TITLE
Refactor demos with oak.js sdk

### PR DESCRIPTION
**Pull request by imstar15**
### mangata-rococo

Task ID: 2238585-2-3

Task was executed on turing-staging:

https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frpc.turing-staging.oak.tech#/explorer/query/0x7e7ca5d482e7ab23e9738cea83f65c6c1ac1cd1bad6b59535ca40e09d9433a56

Task was executed on mangata-rococo:

https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fcollator-01-ws-rococo.mangata.online#/explorer/query/0xdef43a347764493834121560ee30924304cfde836de8b394c46e71bb6fb38fec

### moonbase-alpha

Task ID: 497899-0-1

Task was executed on turing-moonbase:

http://167.99.226.24:3000/?rpc=ws%3A%2F%2F167.99.226.24%3A8846#/explorer/query/0x54f294e93af8dd9ba2857fdc082c0c8b39939b53831197ac4dc97750edd43ce9

Task was executed on moonbase-alpha:

https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fwss.api.moonbase.moonbeam.network#/explorer/query/5058771

### rocstar

Task ID: 2238684-0-1

Task was executed on turing-staging:

https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frpc.turing-staging.oak.tech#/explorer/query/0xd4f2bccb0faf9e8a7ca01d53283b1d456fb9d57c556b4b57114078bcd20bb1cb

Task was executed on rocstar:

https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frocstar.astar.network#/explorer/query/1982580

### CODE REVIEW

File: src/moonbeam/common.js [hunk 1]
```diff
@@ -30,204 +126,61 @@ const sendXcmFromMoonbase = async ({
...
-    console.log(`\na) Add a proxy for Alice If there is none setup on Turing (paraId:${moonbaseHelper.config.paraId})\n`);
-    const proxyTypeTuring = 'Any';
-    const proxiesOnTuring = await turingHelper.getProxies(turingAddress);
-    const proxyMatchTuring = _.find(proxiesOnTuring, { delegate: proxyOnTuring, proxyType: proxyTypeTuring });
-    if (proxyMatchTuring) {
-        console.log(`Proxy address ${proxyOnTuring} for paraId: ${moonbaseHelper.config.paraId} and proxyType: ${proxyTypeTuring} already exists; skipping creation ...`);
-    } else {
-        console.log(`Add a proxy of ${parachainName} (paraId:${moonbaseHelper.config.paraId}) and proxyType: ${proxyTypeTuring} on Turing ...\n Proxy address: ${proxyOnTuring}\n`);
-        await sendExtrinsic(turingHelper.api, turingHelper.api.tx.proxy.addProxy(proxyOnTuring, proxyTypeTuring, 0), keyPair);
+    if (!isTaskExecuted) {
```
Comment (chrisli30) [inline src/moonbeam/common.js:160]: No `!isTaskExecuted`

File: src/moonbeam/moonbase-local.js [hunk 2]
```diff
@@ -1,287 +1,32 @@
...
-        callWeight: transactCallWeight,
-        overallWeight,
-        fee,
-    });
-
-    console.log(`transactExtrinsic Encoded call data: ${transactExtrinsic.method.toHex()}`);
-
-    await sendExtrinsic(parachainHelper.api, transactExtrinsic, keyPair);
-};
+const ALITH_PRIVATE_KEY = '0x5fb92d6e98884f76de468fa3f6278f8807c48bebc13595d45af5bdc4da702133';
```
Comment (chrisli30) [inline src/moonbeam/moonbase-local.js:8]: Add **a code comment above this line** to show the original github link where this private key is already exposed.

File: src/utils.js [hunk 3]
```diff
@@ -171,25 +174,23 @@ async function main() {
...
         console.log(`Restored account ${keyPair.meta.name} ${keyPair.address} ...`);
 
-        const account = new Account(keyPair);
-        await account.init([mangataHelper]);
-        account.print();
-
-        const { pair } = account;
         const targetSellAmount = 100;
-        const minExpectedBuyAmount = 150;
+        // const minExpectedBuyAmount = 150;
```
Comment (chrisli30) [inline src/utils.js:178]: What’s up with this line of code comment? Do we still need it?

**Review by chrisli30 — CHANGES_REQUESTED** [review]

File: src/astar/common.js [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+const MIN_BALANCE_IN_PROXY = 10; // The proxy accounts are to be topped up if its balance fails below this number
+const TASK_FREQUENCY = 3600;
+
+// eslint-disable-next-line import/prefer-default-export
+export const scheduleTask = async ({
+    oakConfig, astarConfig, scheduleActionType, createPayloadFunc, keyringPair,
+}) => {
+    const keyring = new Keyring({ type: 'sr25519' });
+
+    const oakHelper = new OakHelper({ endpoint: oakConfig.endpoint });
```
[the remark this answers is not in the document]
Comment (imstar15) [reply src/astar/common.js:24]: Currently, OakHelper is only used to create oakApi, which involves several steps:

1.` import '@oak-network/api-augment'`
2. Initialize ApiPromise with rpc, types, runtime
```
import { rpc, types, runtime } from '@oak-network/types';
....
this.api = await ApiPromise.create({
      provider: new WsProvider(this.config.endpoint), rpc, types, runtime,
});
```

Therefore, I believe we need to encapsulate these technical details by using a helper.

File: src/astar/common.js [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+    const isTaskExecutedAgain = await listenEvents(astarApi, 'proxy', 'ProxyExecuted', nextExecutionTimeout);
+
+    if (isTaskExecutedAgain) {
+        console.log('Task cancellation failed! It executes again.');
+        return;
+    }
+    console.log("Task canceled successfully! It didn't execute again.");
+
+    await oakHelper.finalize();
+    await astarHelper.finalize();
```
[the remark this answers is not in the document]
Comment (imstar15) [reply src/astar/common.js]: Fixed

File: src/mangata/setup-mangata.js [hunk 6]
```diff
@@ -27,148 +24,141 @@ const keyring = new Keyring({ type: 'sr25519' });
...
-            console.log(`[${account.name}] ${symbol} balance on Mangata is zero; minting ${symbol} with sudo ...`);
-            // Because sending extrinsic in parallel will cause repeated nonce errors
-            // we need to wait for the previous extrinsic to be finalized before sending the extrinsic.
-            await mangataHelper.mintToken(mangataAddress, symbol, account.pair);
-        }
-    }
+    // const proxyAccountId = mangataAdapter.getDerivativeAccount(u8aToHex(keyringPair.addressRaw), oakAdapter.getChainData().paraId);
+    // const proxyAddress = keyring.encodeAddress(proxyAccountId, mangataAdapter.getChainData().ss58Prefix);
+    // const proxiesResponse = await mangataHelper.api.query.proxy.proxies(mangataAddress);
+    // const proxies = _.first(proxiesResponse.toJSON());
```
[the remark this answers is not in the document]
Comment (imstar15) [reply src/mangata/setup-mangata.js]: Fixed

File: src/mangata/setup-mangata.js [hunk 7]
```diff
@@ -27,148 +24,141 @@ const keyring = new Keyring({ type: 'sr25519' });
...
-
-        console.log(`Adding a proxy for paraId ${turingHelper.config.paraId}. Proxy address: ${proxyAddress} ...`);
-        await mangataHelper.addProxy(proxyAddress, proxyType, account.pair);
-    }
+    //     console.log(`Adding a proxy for paraId ${oakAdapter.getChainData().paraId}. Proxy address: ${proxyAddress} ...`);
+    //     await mangataHelper.addProxy(proxyAddress, proxyType, keyringPair);
+    // }
 
     const answerPool = await confirm({ message: '\nAccount setup is completed. Press ENTRE to set up pools.', default: true });
+    if (!answerPool) return;
```
[the remark this answers is not in the document]
Comment (imstar15) [reply src/mangata/setup-mangata.js]: Fixed

File: src/moonbeam/common.js (hunk 1 above)
Comment (imstar15) [reply src/moonbeam/common.js]: Fixed

File: src/utils.js (hunk 3 above)
Comment (imstar15) [reply src/utils.js]: We don't need the variable.
Removed.

File: src/mangata/setup-mangata.js [hunk 8]
```diff
@@ -27,148 +24,141 @@ const keyring = new Keyring({ type: 'sr25519' });
...
+    const liquidityAsset = _.find(assets, { id: pool.liquidityTokenId });
+    console.log(`Found a pool of ${liquidityAsset.symbol}`, pool);
+
+    // Promote pool
+    console.log('\n6. Promote the pool to activate liquidity rewarding ...');
+    await mangataHelper.updatePoolPromotion(pool.liquidityTokenId, 100, keyringPair);
+
+    // Mint liquidity to generate rewards...
+    console.log('\n7. Mint liquidity to generate rewards...');
+    const formatedPool = mangataHelper.formatPool(pool, mangataAsset.decimals, oakAsset.decimals);
```
[the remark this answers is not in the document]
Comment (imstar15) [reply src/mangata/setup-mangata.js]: Fixed

File: src/moonbeam/moonbase-local.js (hunk 2 above)
Comment (imstar15) [reply src/moonbeam/moonbase-local.js:10]: Fixed

File: src/moonbeam/common.js [hunk 9]
```diff
@@ -1,25 +1,121 @@
...
 import {
-    sendExtrinsic, readEthMnemonicFromFile, readMnemonicFromFile, listenEvents, getTaskIdInTaskScheduledEvent, waitPromises,
+    sendExtrinsic, listenEvents, getTaskIdInTaskScheduledEvent, getHourlyTimestamp, waitPromises, ScheduleActionType, calculateTimeout,
 } from '../common/utils';
+import OakHelper from '../common/oakHelper';
+import MoonbeamHelper from '../common/moonbeamHelper';
 
-const CONTRACT_ADDRESS = '0x7Dd212Fd97ab7C5FA3D44031CE1b9b50248e3177';
+const TASK_FREQUENCY = 3600;
+const CONTRACT_ADDRESS = '0xa72f549a1a12b9b49f30a7f3aeb1f4e96389c5d8';
```
[the remark this answers is not in the document]
Comment (imstar15) [reply src/moonbeam/common.js:16]: Fixed

**Review by chrisli30 — APPROVED** [review]
I think this PR is in a good state now. Good to merge!

**Pull request merged**